lightning/src/ln/peer_handler.rs

```diff
@@ -864,7 +873,20 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 									peer.their_node_id = Some(their_node_id);
 									insert_node_id!();
 									let features = InitFeatures::known();
-									let resp = msgs::Init { features };
+									let resp = msgs::Init { features, remote_network_address:  match peer.their_net_address.clone(){
+										Some(NetAddress::IPv4{addr: [10, 0..=255, ..], port: _}) => None,
```

Comment:
Can you link to IANA's IP allocation page in some comments, and also DRY this code? Once you do that, can you add a test for it to check various IPs using `std::net::IpAddress`

Reply:
By DRY, do you want me to make a macro for this code or a function?

Reply:
In general, things should only be macro'd if they cannot be function'd, though there's some places in our code where we don't follow that convention.

Reply:
> Once you do that, can you add a test for it to check various IPs using `std::net::IpAddress`

Do you want me to check for all the ip addresses in the range?

Reply:
You should always write tests that you think best convince you and reviewers that the code is correct, and which will find any subtle edge cases which you think could reasonably regress in the future. Focus and that and no need to ask for us specific design decisions :)

Reply:
> Can you link to IANA's IP allocation page in some comments, and also DRY this code? Once you do that, can you add a test for it to check various IPs using `std::net::IpAddress`

I have added 3 tests for each ip range, two corner cases and one in between. Also I wasn't sure how to use `std::net::IpAddress` in this case, so I didn't use it at all.